lib/truffle/truffle/patching.rb

```diff
@@ -0,0 +1,38 @@
+module Truffle::Patching
+  TRUFFLE_PATCHES_DIRECTORY = File.join(Truffle::Boot.ruby_home, 'lib', 'patches')
+  TRUFFLE_PATCHES           = Dir.glob(File.join(TRUFFLE_PATCHES_DIRECTORY, '**', '*.rb')).
+      select { |path| File.file? path }.
+      map { |path| path[(TRUFFLE_PATCHES_DIRECTORY.size+1)..-4] }.
+      reduce ({}) { |h, v| h.update v => true }
```

Comment:
Ah, `Hash#update` is `Hash#merge!` (don't use `update`, it's an uncommon alias)
`v => true` creates a Hash per iteration.
Measure it if you want.
But anyway I think reduce/inject + modifying inplace the accumulator is an anti-pattern that just adds complexity with extra variables and caring of the return value, for no gain.